eng/scripts/Attest-CPEX-KPIs.ps1

```diff
@@ -0,0 +1,153 @@
+    $productType = $fields["Custom.ProductType"]
+    $url = $triage.url
+
+    if ($productType -eq "Sku") {
+        $productType = "ProductSku"
+    }
+
+    AddAttestationEntry $productServiceTreeId $KPI_ID_Onboarding $Complete $productType $url
+
+    # Maps the current lifecycle stage to the required DATAPLANE KPIs needed to advance to the next lifecycle stages
+    $lifecycleToDataKpis = @{
+        "In Dev" = @($KPI_ID_Data_Private_Preview, $KPI_ID_Data_Public_Preview, $KPI_ID_Data_GA)
+        "Private Preview" = @($KPI_ID_Data_Public_Preview, $KPI_ID_Data_GA)
+        "Public Preview" = @($KPI_ID_Data_GA)
+    }
+
+    if ($dataAttestationStatus -ne "Completed") {
+        if ($dataScope -eq "Yes") {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.DataplaneAttestationStatus" -status "Complete"
+        } else {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.DataplaneAttestationStatus" -status "Not applicable"
+
+            if ($lifecycleToDataKpis.ContainsKey($productLifecycle)) {
+                foreach ($kpiId in $lifecycleToDataKpis[$productLifecycle]) {
+                    AddAttestationEntry $productServiceTreeId $kpiId $NA $productType $url
+                }
+            }
+        }
+    }
+
+    # Maps the current lifecycle stage to the required MANGEMENT PLANE KPIs needed to advance to the next lifecycle stages
+    $lifecycleToMgmtKpis = @{
+        "In Dev" = @($KPI_ID_Mgmt_Private_Preview, $KPI_ID_Mgmt_Public_Preview, $KPI_ID_Mgmt_GA)
+        "Private Preview" = @($KPI_ID_Mgmt_Public_Preview, $KPI_ID_Mgmt_GA)
+        "Public Preview" = @($KPI_ID_Mgmt_GA)
+    }
+
+    if ($mgmtAttestationStatus -ne "Completed") {
+        if ($mgmtScope -eq "Yes") {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.ManagementPlaneAttestationStatus" -status "Complete"
+        } else {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.ManagementPlaneAttestationStatus" -status "Not applicable"
+            if ($lifecycleToMgmtKpis.ContainsKey($productLifecycle)) {
+                foreach ($kpiId in $lifecycleToMgmtKpis[$productLifecycle]) {
+                    AddAttestationEntry $productServiceTreeId $kpiId $NA $productType $url
+                }
+            }
+        }
+    }
+
+}
+
+$releasePlans = Get-ReleasePlansForCPEXAttestation
+
+foreach ($releasePlan in $releasePlans) {
+    $fields = $releasePlan.fields
+    $dataScope = $fields["Custom.DataScope"]
+    $mgmtScope = $fields["Custom.MgmtScope"]
+    $lifecycle = $fields["Custom.ReleasePlanType"]
+    $productServiceTreeId = $fields["Custom.ProductServiceTreeID"]
+    $productType = $fields["Custom.ProductType"]
+    $url = $releasePlan.url
+
+    if ($productType -eq "Sku") {
+        $productType = "ProductSku"
+    }
+    
+    if ($dataScope -eq 'Yes') {
```

Comment:
@MrJustinB release plan attestation is simpler. According to the Lifecycle, the attestation happens